EIPS/eip-3030.md

```diff
@@ -0,0 +1,371 @@
+* Connection #0 to host localhost left intact
+{"error":"Invalid signingRoot: 0xaa1"}
+
+# No Keys Available
+
+## Request
+curl -v -X POST -d '{"signingRoot":"0xb6bb8f3765f93f4f1e7c7348479289c9261399a3c6906685e320071a1a13955c"}' -H 'Content-Type: application/json' localhost:9000/sign/000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
+
+## Response
+Note: Unnecessary use of -X or --request, POST is already inferred.
+*   Trying 127.0.0.1:9000...
+* TCP_NODELAY set
+* Connected to localhost (127.0.0.1) port 9000 (#0)
+> POST /sign/000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 HTTP/1.1
+> Host: localhost:9000
+> User-Agent: curl/7.68.0
+> Accept: */*
+> Content-Type: application/json
+> Content-Length: 84
+>
+* upload completely sent off: 84 out of 84 bytes
+* Mark bundle as not supporting multiuse
+< HTTP/1.1 404 Not Found
+< content-length: 123
+< date: Wed, 30 Sep 2020 02:18:53 GMT
+<
+* Connection #0 to host localhost left intact
+{"error":"Key not found: 000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"}
+
+# Server Error
+
+## Preparation
+## `chmod` both keys directory and file to the octal 311 (-wx--x--x).
+## `chmod` back to 755 to delete them afterwards.
+
+## Request
+curl -v -X POST -d '{"signingRoot":"0xb6bb8f3765f93f4f1e7c7348479289c9261399a3c6906685e320071a1a13955c"}' -H 'Content-Type: application/json' localhost:9000/sign/b7354252aa5bce27ab9537fd0158515935f3c3861419e1b4b6c8219b5dbd15fcf907bddf275442f3e32f904f79807a2a
+
+## Response
+Note: Unnecessary use of -X or --request, POST is already inferred.
+*   Trying 127.0.0.1:9000...
+* TCP_NODELAY set
+* Connected to localhost (127.0.0.1) port 9000 (#0)
+> POST /sign/b7354252aa5bce27ab9537fd0158515935f3c3861419e1b4b6c8219b5dbd15fcf907bddf275442f3e32f904f79807a2a HTTP/1.1
+> Host: localhost:9000
+> User-Agent: curl/7.68.0
+> Accept: */*
+> Content-Type: application/json
+> Content-Length: 84
+>
+* upload completely sent off: 84 out of 84 bytes
+* Mark bundle as not supporting multiuse
+< HTTP/1.1 500 Internal Server Error
+< content-length: 43
+< date: Wed, 30 Sep 2020 02:21:08 GMT
+<
+* Connection #0 to host localhost left intact
+{"error":"Storage error: PermissionDenied"}
+```
+
+## Implementation
+
+Repository Url | Language | Organization | Commentary
+--- | --- | --- | ---
+[BLS Remote Signer](https://github.com/sigp/rust-bls-remote-signer) | Rust | Sigma Prime | Supports proposed specification.
+[Web3signer](https://github.com/PegaSysEng/web3signer) | Java | PegaSys | Supports proposed specification, although with [slightly different methods](https://pegasyseng.github.io/web3signer/web3signer-eth2.html):<br>{`/sign` => `/api/v1/eth2/sign`, `/publicKeys` => `/api/v1/eth2/publicKeys`}.
+
+The Prysm client supports a [Remote Signing Wallet](https://docs.prylabs.network/docs/wallet/remote/), however its API requires using gRPC as transport.
```

Comment:
Prysm supports both gRPC and JSON over HTTP. There's a grpc<->json/http proxy. We will update our documentations